lib/rules/no-tracked-properties-from-args.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const ERROR_MESSAGE =
+  'Do not use this.args to create @tracked properties as this will not be updated if the args change. Instead use a getter.';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow creating @tracked properties from this.args',
+      category: 'Ember Decorator',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-tracked-properties-from-args.md',
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    return {
+      PropertyDefinition(node) {
```

Comment:
Do we still need to support `ClassProperty` as well? @bmish

Reply:
It's not a bad idea to support that too until we drop support for ESLint v7. Context: https://github.com/ember-cli/eslint-plugin-ember/pull/1529

Although I'm not super worried about it if it's a hassle...

Reply:
Thanks! Borrowing the implementation from no-private-routing-service.js, but is there a way to have the tests run against both of these versions?

Reply:
CI runs tests against both ESLint v7 and ESLint v8 so as long as it passes, we should be good.